File: ruby/testcode/authzfailure_025.rb
require_relative 'shared'

class InvoiceRecord
  def self.find_by_id(id)
    { id: id, amount: 9_999, customer: "acme_corp" }
  end
end

def current_graphql_user(ctx)
  ctx[:current_user]
end

# vuln-code-snippet start ruby_authz_graphql_field
def resolve_invoice(obj, args, ctx)
  invoice = InvoiceRecord.find_by_id(args[:id]) # vuln-code-snippet vuln-line ruby_authz_graphql_field
  invoice
end
# vuln-code-snippet end ruby_authz_graphql_field

File: rust/testcode/authzfailure_025.rs
//! CWE-285: Horizontal privilege escalation — any user's private notes accessible by note_id

fn db_get_note(note_id: &str) -> String {
    format!("private_note_content_for_{}", note_id)
}

// vuln-code-snippet start testcodeAuthzfailure025
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let note_id = req.param("note_id");
    let note = db_get_note(&note_id); // vuln-code-snippet target-line testcodeAuthzfailure025
    super::shared::BenchmarkResponse::ok(&note)
}
// vuln-code-snippet end testcodeAuthzfailure025
